### PR TITLE
Fix acceptance tests for `google_workstations_workstation_config` following `vm_tags` field being added

### PR DESCRIPTION
**Pull request by SarahFrench**
Some tests fail in VCR following https://github.com/GoogleCloudPlatform/magic-modules/pull/11015, for example: https://github.com/GoogleCloudPlatform/magic-modules/pull/11074#issuecomment-2197166383

```
    vcr_utils.go:152: Step 1/6 error: Error running pre-apply refresh: exit status 1
        
        Error: Inconsistent dependency lock file
        
        The following dependency selections recorded in the lock file are
        inconsistent with the current configuration:
          - provider registry.terraform.io/hashicorp/google: required by this configuration but no version is selected
        
        To make the initial dependency selections that will initialize the dependency
        lock file, run:
          terraform init

```

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by SarahFrench** [comment]
This appears to be the cause: https://github.com/modular-magician/terraform-provider-google-beta/blob/5f8e8db3e62c684887a754db1ed2a4334eb96777/google-beta/services/workstations/resource_workstations_workstation_config_generated_test.go#L58-L64

File: mmv1/products/workstations/WorkstationConfig.yaml [hunk 1]
```diff
@@ -75,6 +75,12 @@ examples:
       workstation_config_name: 'workstation-config'
       tag_key1: 'tag_key1'
       tag_value1: 'tag_value1'
+      project_id: 'my-new-project'
+      project_name: 'my-new-project'
```
Comment (SarahFrench) [inline mmv1/products/workstations/WorkstationConfig.yaml:79]:
```suggestion
```

File: mmv1/products/workstations/WorkstationConfig.yaml [hunk 2]
```diff
@@ -75,6 +75,11 @@ examples:
       workstation_config_name: 'workstation-config'
       tag_key1: 'tag_key1'
       tag_value1: 'tag_value1'
+      project_id: 'my-new-project'
+      org_id: '123456789'
+    test_vars_overrides:
+      project_id: 'fmt.Sprintf("tf-test-workstation-proj-%s", acctest.RandString(t, 10))'
```
Comment (SarahFrench) [inline mmv1/products/workstations/WorkstationConfig.yaml:81]:
```suggestion
      project_id: 'fmt.Sprintf("tf-test-proj-%s", acctest.RandString(t, 10))'
```

Addressing char limit failures

**Comment by SarahFrench** [comment]
I'm juggling this alongside other tasks, hence iterating on errors via this PR that would have been found if I ran tests locally 😪

**Comment by SarahFrench** [comment]
AH... just realised that the PR added a new project resource unnecessarily? So instead of making that project be used for everything the solution is just to remove that unneeded resource... 😮‍💨

**Comment by BBBmau** [comment]
not sure how i let that one slip by 🤦🏼 thanks for catching this.

File: mmv1/third_party/terraform/services/workstations/resource_workstations_workstation_config_test.go.erb [hunk 3]
```diff
@@ -753,13 +753,13 @@ func testAccWorkstationsWorkstationConfig_update(context map[string]interface{})
 resource "google_tags_tag_key" "tag_key1" {
   provider = "google-beta"
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/workstations/resource_workstations_workstation_config_test.go.erb:754]: After the next tests finish I'll remove these quotes for consistency

File: mmv1/third_party/terraform/services/workstations/go/resource_workstations_workstation_config_test.go.tmpl [hunk 4]
```diff
@@ -1333,22 +1333,22 @@ func TestAccWorkstationsWorkstationConfig_vmTags(t *testing.T) {
...
-  }
+return acctest.Nprintf(`
+data "google_project" "project" {
+  provider = google-beta
+}
+
+resource "google_tags_tag_key" "tag_key1" {
+  provider = google-beta
+  parent = "projects/${data.google_project.project.number}"
+  short_name = "tf_test_tag_key1%{random_suffix}"
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/workstations/go/resource_workstations_workstation_config_test.go.tmpl:1344]: I left short_names unchanged here, as not used in combination with the generated `testAccWorkstationsWorkstationConfig_workstationConfigBasicExample` function.

`testAccWorkstationsWorkstationConfig_update` does need to be changed, as that is used in combination with `testAccWorkstationsWorkstationConfig_workstationConfigBasicExample`

File: mmv1/third_party/terraform/services/workstations/go/resource_workstations_workstation_config_test.go.tmpl [hunk 5]
```diff
@@ -707,8 +707,11 @@ func testAccWorkstationsWorkstationConfig_readinessChecks(context map[string]int
 func TestAccWorkstationsWorkstationConfig_update(t *testing.T) {
 	t.Parallel()
 
+	randString := acctest.RandString(t, 10)
 	context := map[string]interface{}{
-		"random_suffix": acctest.RandString(t, 10),
+		"random_suffix": randString,
+		"project_id":    fmt.Sprintf("tf-test-proj-%s", randString),
+		"org_id":        envvar.GetTestOrgFromEnv(t),
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/workstations/go/resource_workstations_workstation_config_test.go.tmpl:714]: This line missing previously led to requests failing due to bad values:

```
---[ REQUEST ]---------------------------------------
POST /v3/tagKeys?alt=json HTTP/1.1
Host: cloudresourcemanager.googleapis.com
User-Agent: Terraform/1.8.3 (+https://www.terraform.io) Terraform-Plugin-SDK/2.33.0 terraform-provider-google-beta/acc
Content-Length: 69
Content-Type: application/json
Accept-Encoding: gzip

{
 "parent": "organizations/",
 "shortName": "tf_test_tag_key12cebrpuokf"
}
```

File: mmv1/third_party/terraform/services/workstations/resource_workstations_workstation_config_test.go.erb [hunk 6]
```diff
@@ -710,6 +711,7 @@ func TestAccWorkstationsWorkstationConfig_update(t *testing.T) {
 
 	context := map[string]interface{}{
 		"random_suffix": acctest.RandString(t, 10),
+		"org_id":        envvar.GetTestOrgFromEnv(t),
```
Comment (SarahFrench) [inline mmv1/third_party/terraform/services/workstations/resource_workstations_workstation_config_test.go.erb:714]: This line missing previously led to requests failing due to bad values:

```
---[ REQUEST ]---------------------------------------
POST /v3/tagKeys?alt=json HTTP/1.1
Host: cloudresourcemanager.googleapis.com
User-Agent: Terraform/1.8.3 (+https://www.terraform.io) Terraform-Plugin-SDK/2.33.0 terraform-provider-google-beta/acc
Content-Length: 69
Content-Type: application/json
Accept-Encoding: gzip

{
 "parent": "organizations/",
 "shortName": "tf_test_tag_key12cebrpuokf"
}
```

**Comment by SarahFrench** [comment]
@BBBmau this fix is ready now, could you please review?

**Comment by SarahFrench** [comment]
> not sure how i let that one slip by 🤦🏼 thanks for catching this.

I think some missing context that might have contributed is that when a resource has IAM resources generated (and the acc tests for those IAM resources generated) the [first example listed in the YAML](https://github.com/tejal29/magic-modules/blob/3e45788e1e7e848f3bff992395f90790fb6798c2/mmv1/products/workstations/WorkstationConfig.yaml#L66-L77) for that resources is used. So [the IAM related tests failing here](https://github.com/GoogleCloudPlatform/magic-modules/pull/11015#issuecomment-2195645896) were due to the contributor altering the `examples/workstation_config_basic.tf.erb` file in a way that makes all the related tests start failing because it mixed use of `google` and `google-beta`. Once that problem was fixed it revealed that there were some other problems, which are fixed in this PR.

So stuff to bear in mind:
- `Error: Inconsistent dependency lock file` : If you see this in a test failure take a look at what provider is explicitly mentioned in the config of the test
- The first example of a given resources is used to generate IAM resource tests
- Sometimes generated test code is re-used in handwritten tests, so you might get handwritten tests start failing if an example file is changed.

**Review by BBBmau — APPROVED** [review]
it all looks good. Tests are passing! 🎊

**Pull request merged**